### DISCUSSION TITLE
Repeater Changes State

### DISCUSSION THREAD

**Discussion by HMM-MK** [filamentphp/filament#8387] [General]
If repeater fields changes , how to sum to other field outside of repeater and repeater total prices?So, How to get repeater Changes State?

                Card::make() 
                     ->schema([ 
                         Repeater::make('treatmentProducts') 
                             ->relationship() 
                             ->schema([ 
                                 Select::make('product_id') 
                                     ->label('Name') 
                                     ->placeholder('Select Product') 
                                     ->options(Product::all()->pluck('name', 'id')) 
                                     ->filled() 
                                     ->afterStateUpdated(function ($state, callable $set) { 
                                         $product = Product::find($state)->with('unit')->first(); 
                                         if($product){ 
                                             $set('unit', $product->unit->name); 
                                             $set('price', $product->sell_price); 
                                         } 
                                     })->reactive(), 
                                 TextInput::make('qty') 
                                     ->numeric() 
                                     ->minValue(1) 
                                     ->reactive() 
                                     ->debounce(600) 
                                     ->filled() 
                                     ->afterStateUpdated(function (Set $set,Get $get){ 
                                         $total = ($get('qty') ? $get('qty') : 0) * ($get('price') ? $get('price') : 0); 
                                         $set('total_product_price', $total); 
                                     }), 
                                 TextInput::make('unit') 
                                     ->reactive() 
                                     ->disabled() 
                                     ->afterStateHydrated(function (TextInput $component, Get $get) { 
                                        if($get('product_id')){ 
                                         $product = Product::find($get('product_id'))->with('unit')->first(); 
                                         $component->state( $product->unit->name); 
                                        } 
                                     }), 
                                 TextInput::make('price') 
                                     ->numeric() 
                                     ->reactive() 
                                     ->disabled() 
                                     ->afterStateHydrated(function (TextInput $component, Get $get) { 
                                        if($get('product_id')){ 
                                         $product = Product::find($get('product_id'))->with('unit')->first(); 
                                         $component->state( $product->sell_price); 
                                        } 
                                     }), 
                                  TextInput::make('total_product_price') 
                                     ->label('Total') 
                                     ->numeric() 
                                     ->reactive() 
                                     ->readOnly(), 
  
                             ])->columns(5) 
                             ->addActionLabel('Add Products') 
                             ->reorderable(true) 
                             ->orderColumn('id') 
                             ->columnSpan(3) 
                             ->defaultItems(0), 
                         TextInput::make('treatment') 
                             ->required() 
                             ->filled() 
                             ->placeholder('Enter treatment'), 
                         TextInput::make('charge') 
                             ->label('Treatment Cost') 
                             ->numeric() 
                             ->required() 
                             ->reactive() 
                             ->debounce(600) 
                             ->filled() 
                             ->minValue(0) 
                             ->placeholder('Enter treatment cost') 
                             ->helperText("Include Doctor's Fees") 
                             ->afterStateUpdated(function ( $state,Set $set,Get $get) { 
                                 $treatmentProducts = $get('treatmentProducts'); 
                                 $sum = 0; 
                                 foreach ($treatmentProducts as $item) { 
                                     if (isset($item['total_product_price'])) { 
                                         $sum += is_numeric($item['total_product_price']) ? $item['total_product_price'] : 0; 
                                     } 
                                 } 
                                 $set('sub_total', $sum + $state); 
                             }), 
                         TimePicker::make('duration') 
                             ->withoutSeconds() 
                             ->required() 
                             ->filled(), 
                         TextInput::make('sub_total') 
                             ->label('Sub Total') 
                             ->readOnly() 
                             ->filled() 
                             ->reactive() 
                             ->columnStart(3) 
                     ])->columns(3) 
             ]);

**Comment by leandrocfe** [comment]
https://filamentphp.com/docs/3.x/forms/fields/repeater/#using-get-to-access-parent-field-values
